forum/frontend/src/components/register.js

```diff
@@ -28,109 +28,135 @@ const Register = () => {
     return null;
   };
 
-  const handleSubmit = async (e) => {
-    e.preventDefault();
-    
-    const validationErrorMessage = validateInputs();
-    if (validationErrorMessage) {
-      setValidationError(validationErrorMessage);
+  const handleSubmitForm = (username, email, password, firstName, lastName, phone) => {
+    setLoading(true);
+    setError('');
+    setSuccess('');
+
+    const validationError = validateInputs();
+    if (validationError) {
+      setError(validationError);
+      setLoading(false);
       return;
     }
 
-    setLoading(true);
-    
-    try {
-      const response = await axios.post('http://localhost:8000/api/v1/register/', {
-        email: email,
-        password: password,
-        username: username,
-        first_name: firstName,
-        last_name: lastName,
-        phone: phone,
-      });
-
-      console.log('User registered successfully:', response.data);
-      navigate('/login');
-    } catch (error) {
-      if (error.response) {
-        const { data } = error.response;
-        if (data.email) {
-          setError('This email is already in use.');
-        } else if (data.username) {
-          setError('This username is already taken.');
-        } else {
-          setError('Registration failed.');
-        }
-      } else {
-        setError('An error occurred. Please try again.');
+    api.post('/api/v1/register/', {
+      username,
+      email,
+      password,
+      first_name: firstName,
+      last_name: lastName,
+      phone,
+    })
+    .then(response => {
+      if (response.status === 201) {
+        setSuccess('Registration successful! Redirecting to login...');
+        setTimeout(() => navigate('/login'), 2000); // Redirect after 2 seconds
       }
-    } finally {
+    })
+    .catch(error => {
+      const errorMessage = error.response?.data?.non_field_errors ||
+                           error.response?.data?.email ||
+                           error.response?.data?.password ||
+                           'Registration failed. Please try again.';
+      setError(errorMessage);
+    })
+    .finally(() => {
       setLoading(false);
-    }
+    });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    handleSubmitForm(username, email, password, firstName, lastName, phone);
+  };
+
+  const handleGoogleRegistration = () => {
+    const clientID = "442876958174-i2eara8qgmaon227mjd2an0b3egffo48.apps.googleusercontent.com";
+    const callBackURI = "http://127.0.0.1:3000/login/success/";
```

Comment:
Store the Google `clientID` and `callBackURI` in environment variables to avoid exposing sensitive information in the code.

Reply:
I tried to do it to the PR, but the error redirect_uri_mismatch I installed the necessary dependencies, but the error did not go away.